src/core/pointcloud/qgspointcloudlayerrenderer.cpp

```diff
@@ -280,96 +280,74 @@ int QgsPointCloudLayerRenderer::renderNodesAsync( const QVector<IndexedPointClou
   QElapsedTimer downloadTimer;
   downloadTimer.start();
 
-  // Instead of loading all point blocks in parallel and then rendering the one by one,
-  // we split the processing into groups of size groupSize where we load the blocks of the group
-  // in parallel and then render the group's blocks sequentially.
-  // This way helps QGIS stay responsive if the nodes vector size is big
-  const int groupSize = 4;
-  for ( int groupIndex = 0; groupIndex < nodes.size(); groupIndex += groupSize )
+  if ( context.feedback() && context.feedback()->isCanceled() )
+    return 0;
+
+  // Async loading of nodes
+  QVector<QgsPointCloudBlockRequest *> blockRequests;
+  QEventLoop loop;
+  if ( context.feedback() )
+    QObject::connect( context.feedback(), &QgsFeedback::canceled, &loop, &QEventLoop::quit );
```

Comment:
I think there is a memory leak here, if the loop is cancelled while a block is loading, the block doesn't get deleted

Reply:
Good catch, should be fixed in [a47e617](https://github.com/qgis/QGIS/pull/48211/commits/a47e617e5219969ce451e9023be86096a0cb26cd)